### ISSUE TITLE
Cannot use automatic jsx runtime in fresh

### ISSUE THREAD

**Issue by lifeiscontent** [denoland/fresh#382]
I changed my deno.json to:

```json
{
  "$schema": "https://cdn.deno.land/deno/versions/v1.23.2/raw/cli/schemas/config-file.v1.json",
  "compilerOptions": {
    "jsx": "react-jsx",
    "jsxImportSource": "preact"
  },
  "tasks": {
    "start": "deno run -A --watch=static/,routes/ dev.ts"
  },
  "importMap": "./import_map.json"
}
```

but it doesn't work due to several lines in the fresh codebase explicitly setting

```
/** @jsx h */
```

Is there a reason for this? Why not use the automatic runtime always?

**Comment by lucacasonato** [comment]
The automatic runtime configured via deno.json is not supported in Deno Deploy as of now.

**Comment by lifeiscontent** [comment]
@lucacasonato thanks for the info, much appreciated :)

**Comment by uipoet** [comment]
@lucacasonato is there a work-a-round in the meantime? I am working on a React component library that relies on the automatic import of `React` and trying to determine if I need to refactor upstream.

**Comment by tylerbrostrom** [comment]
FYI: automatic JSX runtime support was recently added to esbuild via https://github.com/evanw/esbuild/pull/2349.

That means, at the very least, files inside `islands` _could_ take advantage of the automatic JSX runtime (if fresh were to update its esbuild dependency, and read in the user’s `deno.json`).

**Comment by lucacasonato** [comment]
Deno Deploy now also supports automatic runtime via `deno.json` and via `/** @jsxImportSource */` :)

**Comment by lucacasonato** [comment]
That is to say: automatic JSX runtime coming to a fresh near you soon!

**Comment by tylerbrostrom** [comment]
@lucacasonato if this isn’t already being worked on, I’d be happy to take a stab at it

**Comment by lucacasonato** [comment]
I'm already working on it. Thanks for the offer though! (Currently blocked by an upstream Deno bug)

**Comment by lucacasonato** [comment]
Update: the upstream bug should be fixed in Deno 1.24.3.

**Comment by lucacasonato** [comment]
JSX automatic mode support has landed in #610. It will be part of next weeks' Fresh 1.1.0 release.
